### PR TITLE
Bug 1431085 - Increase max length of PerformanceSignature.extra_options

### PR DESCRIPTION
**Pull request by ionutgoldan**

### CODE REVIEW

File: treeherder/perf/migrations/0011_inc_extra_options_length.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('perf', '0010_fix_signature_uniqueness'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='performancesignature',
+            name='extra_options',
+            field=models.CharField(blank=True, max_length=832),
```
Comment (edmorley) [inline treeherder/perf/migrations/0011_inc_extra_options_length.py]: 832 characters seems a bit long. What's the reasoning for this number? :-)

Comment (ionutgoldan) [reply treeherder/perf/migrations/0011_inc_extra_options_length.py]: The old "extra_properties" field used to have a 1024 length. This is the closest value I could come to.
It allows us to have many options per alert, with a generous length for each option.

File: schemas/performance-artifact.json [hunk 2]
```diff
@@ -82,9 +82,11 @@
                     "type": "array",
                     "title": "Extra options used in running suite",
                     "items": {
-                        "type": "string"
+                        "type": "string",
+                        "maxLength": 45
                     },
-                    "uniqueItems": true
+                    "uniqueItems": true,
+                    "maxItems": 18
```
Comment (wlach) [inline schemas/performance-artifact.json]: IMO we should lower this to something like 8. I don't see any good reason to allow so many extra options (generally treeherder errs on the side of being restrictive when it comes to what you can put in a data schema, which I think helps push people to reduce complexity overall -- a good thing).

Can you add a simple test or two for the edge cases here to `tests/etl/test_perf_schema.py`

Finally, we also need to mirror any changes to mozilla-central: https://searchfox.org/mozilla-central/source/testing/mozharness/external_tools/performance-artifact-schema.json. Generally in-tree tests check the perf data against that schema and then fail if they violate it.

File: treeherder/perf/migrations/0011_inc_extra_options_length.py (hunk 1 above)
Comment (wlach) [reply treeherder/perf/migrations/0011_inc_extra_options_length.py]: If we reduce the limit on the number of extra options to 8, we can reduce this to 8*45=360.

File: treeherder/perf/models.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ class PerformanceSignature(models.Model):
 
     # extra options to distinguish the test (that don't fit into
     # option collection for whatever reason)
-    extra_options = models.CharField(max_length=60, blank=True)
+    extra_options = models.CharField(max_length=832, blank=True)
```
Comment (wlach) [inline treeherder/perf/models.py]: Add a comment for the rationale behind this length (maybe link to the bug)

File: treeherder/perf/migrations/0011_inc_extra_options_length.py (hunk 1 above)
Comment (ionutgoldan) [reply treeherder/perf/migrations/0011_inc_extra_options_length.py]: > If we reduce the limit on the number of extra options to 8, we can reduce this to 8*45=360.
Yep, that's still a reasonable limit.

**Comment by ionutgoldan** [comment]
@wlach I will mirror the changes on mozilla-central. I just filed a separate bug for that.

**Comment by ionutgoldan** [comment]
@edmorley can we land this?
